### ISSUE TITLE
System.NotImplementedException: The method or operation is not implemented.

### ISSUE THREAD

**Issue by ederjbezerra** [nightlybuilds-net/nightly.xam.audiorecorder#3]
I'm experiencing an issue when trying to start recording.

Steps to reproduce:
 
1) Create a new NightlyRecorderService instance:

`NightlyRecorderService rec = new NightlyRecorderService(
      new RecorderSettings
     {
            IosRecorderSettings = new IosAppleLossLess(),
             DroidRecorderSettings = new DroidOggOpus()
       });`

2) Start recording
```    Stream audioStream = await rec.RecordAsync();```

3) This will crash with exception:
** System.NotImplementedException: The method or operation is not implemented.**

Analyzing rec by adding a breakpoint you can see that IsRecording is also "System.NotImplementedException: The method or operation is not implemented"

**Comment by markjackmilian** [comment]
Hi! 
on what platform does this error occur?

**Comment by ederjbezerra** [comment]
> Hi!
> on what platform does this error occur?

Android / Xamarin.Forms 4.7.0.108

On iOS it seems to work(breakpoint shows the instance and its properties as expected), but I'm not able to do a real test because I have a mac mini and there is no mic. But testing on a physical android device I have this issue and I can't even seen the properties of "rec".

**Comment by markjackmilian** [comment]
Strange, I make a test with the version of Forms that you use

**Comment by ederjbezerra** [comment]
@markjackmilian see these two screenshots of same breakpoint on different platforms:

![Captura de Tela 2021-04-01 às 17 46 16](https://user-images.githubusercontent.com/24656439/113352310-71085d00-9312-11eb-9694-af7469f725a4.png)

![Captura de Tela 2021-04-01 às 17 47 16](https://user-images.githubusercontent.com/24656439/113352318-749be400-9312-11eb-9e5f-2c061b54898f.png)

**Comment by markjackmilian** [comment]
can you show me the android project file?

**Comment by ederjbezerra** [comment]
> can you show me the android project file?

<?xml version="1.0" encoding="utf-8"?>
<Project DefaultTargets="Build" ToolsVersion="4.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
  <PropertyGroup>
    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
    <ProjectTypeGuids>{EFBA0AD7-5A72-4C68-AF49-83D382785DCF};{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}</ProjectTypeGuids>
    <ProjectGuid>{94BA3405-8AC9-47A8-90B4-59DBF2D56EF0}</ProjectGuid>
    <OutputType>Library</OutputType>
    <RootNamespace>MasterDetailPageNavigation.Droid</RootNamespace>
    <MonoAndroidAssetsPrefix>Assets</MonoAndroidAssetsPrefix>
    <MonoAndroidResourcePrefix>Resources</MonoAndroidResourcePrefix>
    <AndroidResgenClass>Resource</AndroidResgenClass>
    <AndroidResgenFile>Resources\Resource.designer.cs</AndroidResgenFile>
    <AndroidApplication>True</AndroidApplication>
    <AssemblyName>MasterDetailPageNavigation.Droid</AssemblyName>
    <AndroidManifest>Properties\AndroidManifest.xml</AndroidManifest>
    <TargetFrameworkVersion>v9.0</TargetFrameworkVersion>
    <AndroidHttpClientHandlerType>Xamarin.Android.Net.AndroidClientHandler</AndroidHttpClientHandlerType>
    <NuGetPackageImportStamp />
    <AndroidLinkMode>None</AndroidLinkMode>
  </PropertyGroup>
  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
    <DebugSymbols>true</DebugSymbols>
    <DebugType>portable</DebugType>
    <Optimize>false</Optimize>
    <OutputPath>bin\Debug</OutputPath>
    <DefineConstants>DEBUG;</DefineConstants>
    <ErrorReport>prompt</ErrorReport>
    <WarningLevel>4</WarningLevel>
    <ConsolePause>false</ConsolePause>
<MandroidExtraArgs>MONO_GC_PARAMS=bridge-implementation=old,nursery-size=128‌​m,soft-heap-limit=1024m</MandroidExtraArgs>
  </PropertyGroup>
  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
    <Optimize>true</Optimize>
    <OutputPath>bin\Release</OutputPath>
    <ErrorReport>prompt</ErrorReport>
    <WarningLevel>4</WarningLevel>
    <AndroidUseSharedRuntime>false</AndroidUseSharedRuntime>
    <ConsolePause>false</ConsolePause>
    <AndroidLinkMode>SdkOnly</AndroidLinkMode>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="Xamarin.Forms" Version="4.7.0.1080" />
    <PackageReference Include="Ansuria.XFGloss">
      <Version>1.1.3</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Forms.Maps">
      <Version>4.8.0.1364</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.GooglePlayServices.Maps">
      <Version>71.1610.4</Version>
    </PackageReference>
    <PackageReference Include="Rg.Plugins.Popup">
      <Version>2.0.0.3</Version>
    </PackageReference>
    <PackageReference Include="SkiaSharp">
      <Version>2.80.1</Version>
    </PackageReference>
    <PackageReference Include="SkiaSharp.Views.Forms">
      <Version>2.80.1</Version>
    </PackageReference>
    <PackageReference Include="TouchTracking.Forms">
      <Version>1.1.0</Version>
    </PackageReference>
    <PackageReference Include="DLToolkit.Forms.Controls.ImageCropView">
      <Version>1.0.7</Version>
    </PackageReference>
    <PackageReference Include="Xam.Plugin.Media">
      <Version>5.0.1</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Firebase.Messaging">
      <Version>71.1740.1</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Android.Support.Compat">
      <Version>28.0.0.3</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Essentials">
      <Version>1.5.3.2</Version>
    </PackageReference>
    <PackageReference Include="JimBobBennett.AnimatedProgress">
      <Version>1.1.0</Version>
    </PackageReference>
    <PackageReference Include="Xam.Plugin.SimpleAudioPlayer">
      <Version>1.4.0</Version>
    </PackageReference>
    <PackageReference Include="Plugin.AudioRecorder">
      <Version>1.1.0</Version>
    </PackageReference>
    <PackageReference Include="Plugin.Permissions">
      <Version>6.0.1</Version>
    </PackageReference>
    <PackageReference Include="Plugin.CurrentActivity">
      <Version>2.1.0.4</Version>
    </PackageReference>
    <PackageReference Include="CarouselView.FormsPlugin">
      <Version>5.2.0</Version>
    </PackageReference>
    <PackageReference Include="Com.Airbnb.Xamarin.Forms.Lottie">
      <Version>3.0.4</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.FFImageLoading.Forms">
      <Version>2.4.11.982</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Android.Support.Annotations">
      <Version>28.0.0.3</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.Android.Support.Exif">
      <Version>28.0.0.3</Version>
    </PackageReference>
    <PackageReference Include="Plugin.Toast">
      <Version>2.1.2</Version>
    </PackageReference>
    <PackageReference Include="RestSharp">
      <Version>106.11.7</Version>
    </PackageReference>
    <PackageReference Include="Ascetic.Plugins.Cropper">
      <Version>1.0.5</Version>
    </PackageReference>
    <PackageReference Include="Plugin.FacebookClient">
      <Version>2.2.1</Version>
    </PackageReference>
    <PackageReference Include="nightly.xam.audiorecorder">
      <Version>1.0.0</Version>
    </PackageReference>
  </ItemGroup>
  <ItemGroup>
    <Reference Include="System" />
    <Reference Include="System.Xml" />
    <Reference Include="System.Core" />
    <Reference Include="Mono.Android" />
    <Reference Include="Mono.Android">
      <HintPath>..\..\..\..\..\Library\Frameworks\Xamarin.Android.framework\Versions\10.1.1.0\lib\xamarin.android\xbuild-frameworks\MonoAndroid\v9.0\Mono.Android.dll</HintPath>
    </Reference>
  </ItemGroup>
  <ItemGroup>
    <ProjectReference Include="..\MasterDetailPageNavigation\MasterDetailPageNavigation.csproj">
      <Project>{9A1ED959-56BE-48DB-BD07-545B00242318}</Project>
      <Name>MasterDetailPageNavigation</Name>
    </ProjectReference>
  </ItemGroup>
  <ItemGroup>
    <Compile Include="MainActivity.cs" />
    <Compile Include="Resources\Resource.designer.cs" />
    <Compile Include="Properties\AssemblyInfo.cs" />
    <Compile Include="DialogRenderer.cs" />
    <Compile Include="SplashActivity.cs" />
    <Compile Include="EntryRenderer.cs" />
    <Compile Include="CustomSwitchRenderer.cs" />
    <Compile Include="MyFireMessagingService.cs" />
    <Compile Include="MyFirebaseIIDService.cs" />
    <Compile Include="LoadingPageServiceDroid.cs" />
    <Compile Include="CustomEntryRenderer.cs" />
    <Compile Include="ListViewRenderer.cs" />
    <Compile Include="CustomCollectionViewRenderer.cs" />
  </ItemGroup>
  <ItemGroup>
    <None Include="Resources\AboutResources.txt" />
    <None Include="Properties\AndroidManifest.xml" />
    <None Include="Assets\AboutAssets.txt" />
    <None Include="Resources\network_security_config.xml" />
  </ItemGroup>
  <ItemGroup>
    <Folder Include="Resources\layout\" />
    <Folder Include="Resources\values\" />
    <Folder Include="Resources\xml\" />
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\drawable-hdpi\icon.png" />
    <AndroidResource Include="Resources\drawable-xhdpi\icon.png" />
    <AndroidResource Include="Resources\drawable-xxhdpi\icon.png" />
    <AndroidResource Include="Resources\drawable\reminders.png" />
    <AndroidResource Include="Resources\drawable\todo.png" />
    <AndroidResource Include="Resources\drawable\logoModal.png">
      <SubType></SubType>
      <Generator></Generator>
    </AndroidResource>
    <AndroidResource Include="Resources\drawable\favicon96_96.png">
      <SubType></SubType>
      <Generator></Generator>
    </AndroidResource>
    <AndroidResource Include="Resources\drawable\splash_screen.xml">
      <SubType></SubType>
      <Generator></Generator>
    </AndroidResource>
    <AndroidResource Include="Resources\drawable\receitas_tiny.png">
      <SubType></SubType>
      <Generator></Generator>
    </AndroidResource>
    <AndroidResource Include="Resources\drawable\mapa.png">
      <SubType></SubType>
      <Generator></Generator>
    </AndroidResource>
    <AndroidResource Include="Resources\drawable\galeria.png">
      <SubType></SubType>
      <Generator></Generator>
    </AndroidResource>
    <AndroidResource Include="Resources\drawable\agenda_tiny.png">
      <SubType></SubType>
      <Generator></Generator>
    </AndroidResource>
    <AndroidResource Include="Resources\drawable\menubg.xml">
      <SubType></SubType>
      <Generator></Generator>
    </AndroidResource>
    <AndroidResource Include="Resources\values\styles.xml" />
    <AndroidResource Include="Resources\xml\file_paths.xml">
      <SubType></SubType>
      <Generator></Generator>
    </AndroidResource>
    <AndroidResource Include="Resources\drawable\topbg.png">
      <SubType></SubType>
      <Generator></Generator>
    </AndroidResource>
    <AndroidResource Include="Resources\drawable\bgchat.png">
      <SubType></SubType>
      <Generator></Generator>
    </AndroidResource>
    <AndroidResource Include="Resources\drawable\icon_macho.png">
      <SubType></SubType>
      <Generator></Generator>
    </AndroidResource>
    <AndroidResource Include="Resources\drawable\icon_femea.png">
      <SubType></SubType>
      <Generator></Generator>
    </AndroidResource>
    <AndroidResource Include="Resources\drawable\icon_default.png">
      <SubType></SubType>
      <Generator></Generator>
    </AndroidResource>
    <AndroidResource Include="Resources\drawable\icon_cam.png">
      <SubType></SubType>
      <Generator></Generator>
    </AndroidResource>
    <AndroidResource Include="Resources\drawable\staticmap.png">
      <SubType></SubType>
      <Generator></Generator>
    </AndroidResource>
    <AndroidResource Include="Resources\drawable\amex.png">
      <SubType></SubType>
      <Generator></Generator>
    </AndroidResource>
    <AndroidResource Include="Resources\drawable\elo.png">
      <SubType></SubType>
      <Generator></Generator>
    </AndroidResource>
    <AndroidResource Include="Resources\drawable\mastercard.png">
      <SubType></SubType>
      <Generator></Generator>
    </AndroidResource>
    <AndroidResource Include="Resources\drawable\diners.png">
      <SubType></SubType>
      <Generator></Generator>
    </AndroidResource>
    <AndroidResource Include="Resources\drawable\visa.png">
      <SubType></SubType>
      <Generator></Generator>
    </AndroidResource>
    <AndroidResource Include="Resources\drawable\hipercard.png">
      <SubType></SubType>
      <Generator></Generator>
    </AndroidResource>
    <AndroidResource Include="Resources\drawable\bandeirascartao.png">
      <SubType></SubType>
      <Generator></Generator>
    </AndroidResource>
    <AndroidResource Include="Resources\values\strings.xml">
      <SubType></SubType>
      <Generator></Generator>
    </AndroidResource>
    <AndroidResource Include="Resources\drawable\background.xml">
      <SubType></SubType>
      <Generator></Generator>
    </AndroidResource>
  </ItemGroup>
  <ItemGroup>
    <AndroidResource Include="Resources\layout\Tabbar.axml" />
    <AndroidResource Include="Resources\layout\Toolbar.axml" />
    <AndroidResource Include="Resources\values\colors.xml" />
  </ItemGroup>
  <ItemGroup>
    <AndroidAsset Include="Assets\Font-Awesome-Free-regular.otf" />
    <AndroidAsset Include="Assets\MyriadPro-Bold.otf" />
    <AndroidAsset Include="Assets\MyriadPro-Regular.otf" />
    <AndroidAsset Include="Assets\Font-Awesome-Free-Solid.otf" />
    <AndroidAsset Include="Assets\Font-Awesome-Free-Solid-900.otf" />
    <AndroidAsset Include="Assets\Font-Awesome-Pro-Regular.otf" />
    <AndroidAsset Include="Assets\carregando_itens.json" />
    <AndroidAsset Include="Assets\wifi_white.json" />
  </ItemGroup>
  <ItemGroup>
    <GoogleServicesJson Include="google-services.json" />
  </ItemGroup>
  <ItemGroup>
    <AndroidEnvironment Include="Properties\enviroment.txt" />
  </ItemGroup>
  <Import Project="$(MSBuildExtensionsPath)\Xamarin\Android\Xamarin.Android.CSharp.targets" />
</Project>

**Comment by ederjbezerra** [comment]
> can you show me the android project file?

Sure... it's here:

[MasterDetailPageNavigation.Droid.csproj.zip](https://github.com/nightlybuilds-net/nightly.xam.audiorecorder/files/6246506/MasterDetailPageNavigation.Droid.csproj.zip)

Sorry, I had close the issue by mistake.

**Comment by markjackmilian** [comment]
OK, I think I got it. 5 min and I upload a new NuGet version

**Comment by markjackmilian** [comment]
Can you try to update nugget package to 1.0.1 version?
Please clean bin and obj and try again..

**Comment by ederjbezerra** [comment]
> Can you try to update nugget package to 1.0.1 version?
> Please clean bin and obj and try again..

Sure, man. Thank you for all this attention you're giving.  I give you a feedback in some minutes.

**Comment by ederjbezerra** [comment]
Sorry to say but now I have this when I open the page which have the plugin:

Could not load type of field 'MasterDetailPageNavigation.XAML.ChatPage:gravador' (9) due to: Could not resolve type with token 01000196 from typeref (expected class 'nightly.xam.audiorecorder.NightlyRecorderService' in assembly 'nightly.xam.audiorecorder, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null') assembly:nightly.xam.audiorecorder, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null type:nightly.xam.audiorecorder.NightlyRecorderService member:(null)

Then the app is terminated.

**Comment by markjackmilian** [comment]
You can try to update Just to make a test?
Update 
<TargetFrameworkVersion>v9.0</TargetFrameworkVersion> 
to V10.0

I start a test project with version 9 to make a test

**Comment by ederjbezerra** [comment]
I can't find that version, the most recent is 1.0.1

  GET https://api.nuget.org/v3/registration5-gz-semver2/nightly.xam.audiorecorder/index.json
  OK https://api.nuget.org/v3/registration5-gz-semver2/nightly.xam.audiorecorder/index.json 866ms
No package updates are available from the current package source for project 'MasterDetailPageNavigation.Droid'.

**Comment by markjackmilian** [comment]
update 
`<TargetFrameworkVersion>v9.0</TargetFrameworkVersion>`
in android csproj
to
`<TargetFrameworkVersion>v10.0</TargetFrameworkVersion>`

the error seems to be that the wrong assembly is referenced because of the android target framework (i've always used android 10)

**Comment by ederjbezerra** [comment]
Won't this make update all my packages, xamarin.forms etc? I have some packages that doesn't works on android 10.

**Comment by markjackmilian** [comment]
Ok, just wait for my test.

**Comment by ederjbezerra** [comment]
> Ok, just wait for my test.

Sure... I wait.

**Comment by markjackmilian** [comment]
ok, can you give a try to version 1.0.2 of nuget package?
Remember to clear bin and obj.
I go to bed, tomorrow morning I will look again

**Comment by ederjbezerra** [comment]
> ok, can you give a try to version 1.0.2 of nuget package?
> Remember to clear bin and obj.
> I go to bed, tomorrow morning I will look again

Hey, just updated to 1.0.2. Now it opens the page but when start recording, it rises this exception:

nightly.xam.audiorecorder.Exceptions.RecorderException: See NativeException property
  at nightly.xam.audiorecorder.NightlyRecorderService.StartRecorder () [0x000aa] in <0573239da41e4b07848c938f923495d7>:0
  at nightly.xam.audiorecorder.NightlyRecorderService.RecordAsync () [0x00009] in <0573239da41e4b07848c938f923495d7>:0
  at MasterDetailPageNavigation.XAML.ChatPage.SendMsgTapped (System.Object sender, System.EventArgs e) [0x002df] in /Users/eder/Documents/Projeto Teste/Arquivo Comprimido/MasterDetailPageNavigation/XAML/ChatPage.xaml.cs:374
  at System.Runtime.CompilerServices.AsyncMethodBuilderCore+<>c.<ThrowAsync>b__7_0 (System.Object state) [0x00000] in /Users/builder/jenkins/workspace/archive-mono/2020-02/android/release/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/AsyncMethodBuilder.cs:1021
  at Android.App.SyncContext+<>c__DisplayClass2_0.<Post>b__0 () [0x00000] in <9a14097a65a445eaa85f6a3a1ade52a3>:0
  at Java.Lang.Thread+RunnableImplementor.Run () [0x00008] in <9a14097a65a445eaa85f6a3a1ade52a3>:0
  at Java.Lang.IRunnableInvoker.n_Run (System.IntPtr jnienv, System.IntPtr native__this) [0x00008] in <9a14097a65a445eaa85f6a3a1ade52a3>:0
  at at (wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.1(intptr,intptr)

**Comment by markjackmilian** [comment]
Hi, I think is a permission issue. For Android target framework < 10 permission must be explicit requested.
I'll manage permission inside the library in next releases.
I saw that you have Xamarin.Essentials in your project.
Can you try something like this before call StartRecord() method?

```
   var status = await Permissions.CheckStatusAsync<Permissions.Microphone>();

            if (status != PermissionStatus.Granted)
            {
                status = await Permissions.RequestAsync<Permissions.Microphone>();
                if (status != PermissionStatus.Granted)
                {
                    await this.DisplayAlert("Alert","you need pemrission to mic", "ok");
                    return;
                }
            }
```

and add this in your AndroidManifest:

 `<uses-permission android:name="android.permission.RECORD_AUDIO" />`

Let me know.

**Comment by ederjbezerra** [comment]
@markjackmilian I'm almost sure that's not permission related. I already have a recording plugin working(but records only in wav), so I'm doing the permission request before start recording and my android manifest already have that line.

I have something like this:

```
Permissoes permissoes = new Permissoes();
if (await permissoes.MicPermission())
    audioStream = gravador.RecordAsync();
```

Permission code:

```
public async Task<bool> MicPermission()
        {
            try
            {
                var status = await CrossPermissions.Current.CheckPermissionStatusAsync(Plugin.Permissions.Abstractions.Permission.Microphone);
                if (status == Plugin.Permissions.Abstractions.PermissionStatus.Granted)
                    return true;
                else
                {
                    if (await CrossPermissions.Current.ShouldShowRequestPermissionRationaleAsync(Plugin.Permissions.Abstractions.Permission.Microphone))
                    {
                        await App.Current.MainPage.DisplayAlert("Microfone", "Por favor, dê permissão para o app iniciar o microfone.", "OK");
                    }

                    await CrossPermissions.Current.RequestPermissionsAsync(Plugin.Permissions.Abstractions.Permission.Microphone);
                    status = await CrossPermissions.Current.CheckPermissionStatusAsync(Plugin.Permissions.Abstractions.Permission.Microphone);
                    if (status == Plugin.Permissions.Abstractions.PermissionStatus.Granted)
                        return true;
                    else
                        return false;
                }
            }
            catch (Exception ex)
            {
                Console.WriteLine("Deu um erro ao pegar a permissão do microfone =========>" + ex.Message);
                return false;
            }
        }
```

If I change nightly plugin by xamarin.audiorecorder it works but files are huge, that's why I really need your plugin.
Sorry for the delay, I was sleeping, I'm in Brazil so we have a difference of 3~4 hours.

**Comment by markjackmilian** [comment]
I'll share with you a working example project with TargetFramework 9.0
[App1.zip](https://github.com/nightlybuilds-net/nightly.xam.audiorecorder/files/6249942/App1.zip)

Can you show me the detail of the exception of your not working project?
nightly.xam.audiorecorder.Exceptions.RecorderException: See NativeException property
You'll find a NativeException property with a JavaException.

**Comment by ederjbezerra** [comment]
This is NativeException:
```
{Java.Lang.RuntimeException: start failed.
  at Java.Interop.JniEnvironment+InstanceMethods.CallVoidMethod (Java.Interop.JniObjectReference instance, Java.Interop.JniMethodInfo method, Java.Interop.JniArgumentValue* args) [0x0006e] in <42748fcc36b74733af2d9940a…}
```
I tried your code for requesting permissions:

``` var status = await Permissions.CheckStatusAsync<Permissions.Microphone>(); ```

But same problem.

Your zipped project test works but mine is almost exactly how it is.

**Comment by ederjbezerra** [comment]
StackTrace

"  at Java.Interop.JniEnvironment+InstanceMethods.CallVoidMethod (Java.Interop.JniObjectReference instance, Java.Interop.JniMethodInfo method, Java.Interop.JniArgumentValue* args) [0x0006e] in <42748fcc36b74733af2d9940a8f3cc8e>:0 \n  at Java.Interop.JniPeerMembers+JniInstanceMethods.InvokeVirtualVoidMethod (System.String encodedMember, Java.Interop.IJavaPeerable self, Java.Interop.JniArgumentValue* parameters) [0x0002a] in <42748fcc36b74733af2d9940a8f3cc8e>:0 \n  at Android.Media.MediaRecorder.Start () [0x0000a] in <9a14097a65a445eaa85f6a3a1ade52a3>:0 \n  at nightly.xam.audiorecorder.NightlyRecorderService.StartRecorder () [0x00088] in <0573239da41e4b07848c938f923495d7>:0 \n  --- End of managed Java.Lang.RuntimeException stack trace ---\njava.lang.RuntimeException: start failed.\n\tat android.media.MediaRecorder._start(Native Method)\n\tat android.media.MediaRecorder.start(MediaRecorder.java:1175)\n\tat mono.android.animation.ValueAnimator_AnimatorUpdateListenerImplementor.n_onAnimationUpdate(Native Method)\n\tat mono.android.animation.ValueAnimator_AnimatorUpdateListenerImplementor.onAnimationUpdate(ValueAnimator_AnimatorUpdateListenerImplementor.java:30)\n\tat android.animation.ValueAnimator.animateValue(ValueAnimator.java:1346)\n\tat android.animation.ValueAnimator.animateBasedOnTime(ValueAnimator.java:1258)\n\tat android.animation.ValueAnimator.doAnimationFrame(ValueAnimator.java:1306)\n\tat android.animation.AnimationHandler.doAnimationFrame(AnimationHandler.java:146)\n\tat android.animation.AnimationHandler.-wrap2(AnimationHandler.java)\n\tat android.animation.AnimationHandler$1.doFrame(AnimationHandler.java:54)\n\tat android.view.Choreographer$CallbackRecord.run(Choreographer.java:925)\n\tat android.view.Choreographer.doCallbacks(Choreographer.java:702)\n\tat android.view.Choreographer.doFrame(Choreographer.java:635)\n\tat android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:913)\n\tat android.os.Handler.handleCallback(Handler.java:751)\n\tat android.os.Handler.dispatchMessage(Handler.java:95)\n\tat android.os.Looper.loop(Looper.java:154)\n\tat android.app.ActivityThread.main(ActivityThread.java:6682)\n\tat java.lang.reflect.Method.invoke(Native Method)\n\tat com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1520)\n\tat com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1410)\n"

**Comment by markjackmilian** [comment]
I had that error because of permission.
Can you share a project with the same error you have?
so I could do more accurate tests

**Comment by ederjbezerra** [comment]
> I had that error because of permission.
> Can you share a project with the same error you have?
> so I could do more accurate tests

Yes, I will try that. I'm finishing of having lunch and send it to you.

**Comment by ederjbezerra** [comment]
Do you have whatsapp? we can talk in a fast way and then put the final solution here.

**Comment by markjackmilian** [comment]
I use telegram.. do you have telegram?

**Comment by ederjbezerra** [comment]
+55 83 998225491

**Comment by markjackmilian** [comment]
the problem occurred on project android version with targetframework 9. The problem has been fixed since version 1.0.2

thanks for the report

**Comment by ederjbezerra** [comment]
Fixed! Thank you.
